src/lib/ptree/json.lua

```diff
@@ -1,7 +1,22 @@
 -- Use of this source code is governed by the Apache 2.0 license; see COPYING.
 module(..., package.seeall)
 
--- A very limited json library that only does objects of strings.
+-- A slightly-less-limited json library that should handle all JSON syntax.
+-- Tables are encoded to a JSON array if all their keys are valid integers.
+-- NOTE: Integer-indexed sparse tables are encoded as lists as well! Be careful.
+
+-- nil is represented as an empty table 
+local NIL = {}
+local constants = {
+   ["true"]  = true, 
+   ["false"] = false, 
+   ["null"]  = NIL 
+}
+
+-- Forward declarations as some of these functions call recursively
+local read_json_string, read_json_array, read_json_number_or_bool, read_json_type
+local write_json_null, write_json_scalar, write_json_string, write_json_type
```

Comment:
`read_json_number_or_bool` is not defined. The only functions that are recursively dependent are `read_json_string` and `read_json_type`. For all the other functions that are not recursively dependent they should be declared following the same style as the rest of the code base, that means, `function fn(args)` instead of `fn = function(args)`.

Reply:
Yep, laziness on my behalf here trying to avoid reorganising all the functions 😅 This also affects `write_json_type` and `write_json_string` so I've kept those the same, reorganised and switched the others back to the previous syntax.